transcom/mymove
# MB-2537 Fix body-schema discrepancy with Make MTO Available to Prime endpoint

## Description

The support endpoint to make an MTO available to the Prime doesn't take in any parameters other than the `moveTaskOrderID`, which is a path parameter, and `ifMatch`, which is a header. Therefore, we should remove the body as a required parameter because it isn't actually used. 

Additionally, this endpoint has been renamed from `updateMoveTaskOrderStatus` to `makeMoveTaskOrderAvailable`, which is a more accurate description of this endpoint's functionality. 

## Setup

You need an ID and an eTag for a non-Prime move task order to test this endpoint. This means you can't easily get this info by using the `fetch-mto-updates` endpoint (which only returns MTOs that are available to Prime). There are two main ways to get around this:

1. Connect to the GHC API to grab non-Prime Move Task Order IDs and eTags
2. Grab the info you need from a Prime-available MTO using `fetch-mto-updates`, then go into the database directly and toggle the value for `is_available_to_prime` to be false.

### Test with the GHC API in Postman

To connect to the GHC API, you'll need to grab the office session cookie to authenticate yourself. These are the steps:

1. In Postman, set up an Office Environment with the following values:

| variable | initial value | current value |
| --- | --- | --- |
| baseUrl | http://officelocal:3000/ghc/v1 | http://officelocal:3000/ghc/v1 |

2. Run the server and office client in Chrome:

```bash
$ make server_run
$ make office_client_run
```

3. Open `officelocal:3000` and click Local Sign In
4. Click Login next to any of the recent users. (This populates two cookies in Postman if you are using Interceptor).
5. Add the `office_session_token` manually: View cookies in your browser's dev tools and copy the value for the `office_session_token`.
6. In Postman app, click 'Cookies' underneath the Send button > Click 'Add cookie' under the `officelocal` domain
7. Paste in `office_session_token={value}`
8. Send a GET request to `{{baseUrl}}/move-orders/` and grab a move order ID
9. Send a GET request to `{{baseUrl}}/move-orders/:moveOrderID/move-task-orders` and grab the ID and eTag for a move task order that isn't available to Prime.
10. Call the `makeMoveTaskOrderAvailable` endpoint with these values using either Postman or the Prime API Client (or both). Make sure to omit the body.

### Test with the local DB directly

Open up your favorite database viewer and go for it!

General Prime & Support API instructions: https://github.com/transcom/prime_api_deliverable/wiki/API-Endpoints

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-2537) for this change
